### PR TITLE
[APPAI-1418] Pin the dependent version to master commit hash for better stability

### PR DESCRIPTION
**Pull request by dgpatelgit**
# Description

Pinning the dependent version to master code for better stability

Fixes # (issue)
Jira Issue :: https://issues.redhat.com/browse/APPAI-1418

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -6,7 +6,7 @@ requests
 beautifulsoup4
 numpy
 scipy
-git+https://github.com/fabric8-analytics/nvdlib.git#egg=nvdlib
-git+https://github.com/fabric8-analytics/fabric8-analytics-nvd-toolkit.git#egg=toolkit
-git+https://github.com/fabric8-analytics/fabric8-analytics-utils.git#egg=fabric8-analytics-utils
-git+https://github.com/fabric8-analytics/fabric8-analytics-version-comparator.git#egg=f8a_version_comparator
+git+https://github.com/fabric8-analytics/nvdlib.git@89a9e7e#egg=nvdlib
```
Comment (prashbnair) [inline requirements.txt:9]: Why is this needed for libraries that you have not updated?

File: tests/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,4 @@
 pytest
-pytest-mock
\ No newline at end of file
+pytest-mock
+billiard
```
Comment (prashbnair) [inline tests/requirements.txt:3]: Why are these extra packages required?

File: requirements.txt (hunk 1 above)
Comment (prashbnair) [inline requirements.txt:9]: Why is this needed for packages you have not updated?

File: tests/requirements.txt (hunk 2 above)
Comment (prashbnair) [inline tests/requirements.txt:3]: Why are these extra packages needed?

File: requirements.txt (hunk 1 above)
Comment (dgpatelgit) [reply requirements.txt:9]: @prashbnair We are doing to resolve dependency vulnerability from all CRDA component in phase manner, So we will modify other dependent component as well in near sprints, that's why I pinned all dependent component. Also I have noticed that these component are not changed since many months and don't see any active work also. Felt very safe to pin them to current latest. If you feel otherwise let me know.

File: tests/requirements.txt (hunk 2 above)
Comment (dgpatelgit) [reply tests/requirements.txt:3]: Same as above.

File: tests/requirements.txt [hunk 3]
```diff
@@ -1,2 +1,4 @@
...
+amqp
```
Comment (arajkumar) [inline tests/requirements.txt:4]: I don't see any usage reference to amqp & billiard.

Comment (dgpatelgit) [reply tests/requirements.txt:4]: It was required during testing, I saw failure for this dependencies so added them.

Comment (arajkumar) [reply tests/requirements.txt:4]: Can you share the failure message here? Just want to understand it.

File: tests/requirements.txt (hunk 2 above)
Comment (dgpatelgit) [reply tests/requirements.txt]: Removed extra packages.

File: tests/requirements.txt (hunk 3 above)
Comment (dgpatelgit) [reply tests/requirements.txt]: When I was verifying now without this packages, all checks have passed. Removing this packages from test dependencies.

File: requirements.txt (hunk 1 above)
Comment (prashbnair) [reply requirements.txt:9]: I am not convinced of this approach. I am approving this for now but we need to have a discussion around this.
